### ISSUE TITLE
Storing private keys on wallet?

### ISSUE THREAD

**Issue by MattHalloran** [Berry-Pool/nami-wallet#118]
I'm trying to wrap my head around how to use Nami to authenticate users for an application which also stores off-chain, private user data. I posted the question on [StackExchange](https://cardano.stackexchange.com/questions/4793/limitations-of-wallet-login-for-application-with-off-chain-user-data), but thought it'd be good to ask here as well. 

The problem is that the backend server must verify that a user has access to their wallet. Following the guide on this repo, I was not able to figure out how to implement this. If it is not possible right now, one solution could be to establish a handshake for generating a private key, which can be stored on the wallet and sent to the backend for verification. This would essentially act like a password for traditional sites, and thus Nami would perform similarly to a password manager.

Another solution is to skip centralized storage of private data altogether, and put it on the user's local file system instead. This has its own downsides, mainly that the API for reading local files is not widely supported, and it is difficult to build multi-device support.

In other words, **what are the limitations to using Nami wallet for user authentication, and how can they be solved?**

**Comment by MattHalloran** [comment]
Answered sufficiently [here](https://cardano.stackexchange.com/a/4794/965). Nami provides a signData function, which can be used authenticate with backend.
